### ISSUE TITLE
Typing for SysLogHandler has several errors

### ISSUE THREAD

**Issue by nickwilliams-eventbrite** [python/typeshed#3244]
Using the `logging.handler.SysLogHandler` directly, I get several errors:

```
pysoa/common/logging.py:287: error: Argument 3 to "__init__" of "SysLogHandler" has incompatible type "Optional[int]"; expected "int"
pysoa/common/logging.py:293: error: "SyslogHandler" has no attribute "unixsocket"
pysoa/common/logging.py:293: error: "SyslogHandler" has no attribute "socktype"
pysoa/common/logging.py:332: error: "SyslogHandler" has no attribute "ident"
pysoa/common/logging.py:337: error: "SyslogHandler" has no attribute "facility"
```

All of these are incorrect. According to the source code:

- The `socktype` argument (argument 3) is defined as `socktype=None`
- The constructor sets `socktype = socket.SOCK_DGRAM` if `socktype is None`
- The constructor sets `self.unixsocket`, `self.socktype`, and `self.facility`
- `ident` is a class attribute defined on the class as follows:
  ```
  ident = ''          # prepended to all messages
  ```

**Comment by srittau** [comment]
Patches welcome! But please note that these fields should be commented with `# undocumented`, since they are [not documented](https://docs.python.org/3/library/logging.handlers.html).

**Comment by radkujawa** [comment]
also `priority_names`, `facility_names` and `priority_map` are missing

**Comment by hauntsaninja** [comment]
The type for `__init__` was fixed in #4196, but the field changes are still relevant.
